crates/iota-transactional-test-runner/src/test_adapter.rs

```diff
@@ -1245,6 +1266,97 @@ impl IotaTestAdapter {
+            .map(|arg| arg.into_call_arg(self))
+            .collect::<anyhow::Result<Vec<CallArg>>>()?;
+
+        let ptb_cmds = {
+            let file = ptb_data
+                .ok_or_else(|| anyhow::anyhow!("Missing commands for programmable transaction"))?;
+            let contents = std::fs::read_to_string(file.path())?;
+            let commands = ParsedCommand::parse_vec(&contents)?;
+            let staged = &self.staged_modules;
+            let state = &self.compiled_state;
+            commands
+                .into_iter()
+                .map(|c| {
+                    c.into_command(
+                        &|p| {
+                            let modules = staged
+                                .get(&Symbol::from(p))?
+                                .modules
+                                .iter()
+                                .map(|m| {
+                                    let mut buf = vec![];
+                                    m.module
+                                        .serialize_with_version(m.module.version, &mut buf)
+                                        .unwrap();
+                                    buf
+                                })
+                                .collect();
+                            Some(modules)
+                        },
+                        &|s| Some(state.resolve_named_address(s)),
+                    )
+                })
+                .collect::<anyhow::Result<Vec<Command>>>()?
+        };
+        Ok((ptb_inputs, ptb_cmds))
+    }
+
+    /// Build a MoveAuthenticator.
+    /// Returns the Abstract Test Account and MoveAuthenticator.
+    fn prepare_move_authenticator_data(
+        &mut self,
+        authenticator_inputs: Vec<ParsedValue<IotaExtraValueArgs>>,
+        account: ParsedValue<IotaExtraValueArgs>,
+    ) -> anyhow::Result<(TestAccount, GenericSignature)> {
+        // Resolve authenticator inputs
+        let auth_inputs_resolved = self.compiled_state().resolve_args(authenticator_inputs)?;
+        let auth_inputs: Vec<CallArg> = auth_inputs_resolved
+            .into_iter()
+            .map(|arg| arg.into_call_arg(self))
+            .collect::<anyhow::Result<_>>()?;
+
+        let aa_arg = self
+            .compiled_state()
+            .resolve_args(vec![account])?
+            .into_iter()
+            .next()
+            .ok_or_else(|| anyhow::anyhow!("Missing account for MoveAuthenticator"))?;
+        let aa_arg = aa_arg.into_call_arg(self)?;
+
+        let aa_shared_objects = aa_arg.shared_objects();
+        let aa_shared_object = aa_shared_objects.first().ok_or_else(|| {
+            anyhow::anyhow!(
+                "abstract: account must be a shared object representing the abstract account"
+            )
+        })?;
+        let aa_sender_addr = aa_shared_object.id.into();
+
+        let account = TestAccount {
+            address: aa_sender_addr,
+            key_pair: None,
+            gas: aa_sender_addr.into(),
```

Comment:
This change would avoid confusion for other devs:
```suggestion
            gas: ObjectID::random(),
```

Reply:
You could also add a comment that says `TODO: handle abstract account gas`

Reply:
It will fail the tests

Reply:
I'll leave a TODO